triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRuleSimulationLibTest.scala

```diff
@@ -66,142 +79,482 @@ class TriggerRuleSimulationLibTest
 
   "Trigger rule simulation" should {
     "correctly log metrics for initState lambda" in {
-      forAll(initState) { acs =>
-        for {
-          client <- ledgerClient()
-          party <- allocateParty(client)
-          (trigger, simulator) = getSimulator(
-            client,
-            QualifiedName.assertFromString("Cats:feedingTrigger"),
-            packageId,
-            applicationId,
-            compiledPackages,
-            config.participants(ParticipantId).apiServer.timeProviderType,
-            triggerRunnerConfiguration,
-            party,
-          )
-          (submissions, metrics, state) <- simulator.initialStateLambda(acs)
-        } yield {
-          metrics.evaluation.submissions should be(submissions.size)
-          metrics.submission.creates should be(submissions.map(numberOfCreateCommands).sum)
-          metrics.submission.createAndExercises should be(
-            submissions.map(numberOfCreateAndExerciseCommands).sum
-          )
-          metrics.submission.exercises should be(submissions.map(numberOfExerciseCommands).sum)
-          metrics.submission.exerciseByKeys should be(
-            submissions.map(numberOfExerciseByKeyCommands).sum
-          )
-          metrics.evaluation.steps should be(
-            metrics.evaluation.getTimes + metrics.evaluation.submissions + 1
-          )
-          metrics.startState.acs.activeContracts should be(acs.size)
-          metrics.startState.acs.pendingContracts should be(0)
-          metrics.startState.inFlight.commands should be(0)
-          Some(metrics.endState.acs.activeContracts) should be(
-            numberOfActiveContracts(state, trigger.level, trigger.version)
-          )
-          Some(metrics.endState.acs.pendingContracts) should be(
-            numberOfPendingContracts(state, trigger.level, trigger.version)
-          )
-          Some(metrics.endState.inFlight.commands) should be(
-            numberOfInFlightCommands(state, trigger.level, trigger.version)
-          )
+      for {
+        client <- ledgerClient()
+        party <- allocateParty(client)
+        (trigger, simulator) = getSimulator(
+          client,
+          QualifiedName.assertFromString("Cats:feedingTrigger"),
+          packageId,
+          applicationId,
+          compiledPackages,
+          config.participants(ParticipantId).apiServer.timeProviderType,
+          triggerRunnerConfiguration,
+          party,
+        )
+        result <- forAll(initState) { acs =>
+          for {
+            (submissions, metrics, state) <- simulator.initialStateLambda(acs)
+          } yield {
```

Comment:
Ad hoc timing data suggested that this style of test setup would be more efficient